### PR TITLE
Set collation on database.yml (closes #672)

### LINKED ISSUE #672: MySQL database collation needs to be set on database.yml
Test from docker workbench.

docker-compose run --rm peatio "./bin/setup"

database.yml
default: &default
  adapter: mysql2
  encoding: utf8
  pool: 5
  host: <%= ENV['DATABASE_HOST'] %>
  username: <%= ENV['DATABASE_USER'] %>
  password: <%= ENV['DATABASE_PASS'] %>

  SQL (0.7ms)  INSERT INTO `currencies` (`id`, `code`, `key`, `symbol`, `base_factor`, `quick_withdraw_limit`, `options`, `created_at`, `updated_at`) VALUES (2, 'btc', 'satoshi', '฿', 100000000, 0.1, '{\"api_client\":\"BTC\",\"json_rpc_endpoint\":\"http://user:password@127.0.0.1:18332\",\"bitgo_test_net\":true,\"bitgo_wallet_id\":null,\"bitgo_wallet_address\":null,\"bitgo_wallet_passphrase\":null,\"bitgo_rest_api_root\":\"https://test.bitgo.com/api/v2\",\"bitgo_rest_api_access_token\":null,\"wallet_url_template\":\"https://blockchain.info/address/#{address}\",\"transaction_url_template\":\"https://blockchain.info/tx/#{txid}\"}', '2018-03-10 09:28:39', '2018-03-10 09:28:39')
   (2.2ms)  ROLLBACK
rake aborted!
ActiveRecord::StatementInvalid: Mysql2::Error: Incorrect string value: '\xE0\xB8\xBF' for column 'symbol' at row 1: INSERT INTO `currencies` (`id`, `code`, `key`, `symbol`, `base_factor`, `quick_withdraw_limit`, `options`, `created_at`, `updated_at`) VALUES (2, 'btc', 'satoshi', '฿', 100000000, 0.1, '{\"api_client\":\"BTC\",\"json_rpc_endpoint\":\"http://user:password@127.0.0.1:18332\",\"bitgo_test_net\":true,\"bitgo_wallet_id\":null,\"bitgo_wallet_address\":null,\"bitgo_wallet_passphrase\":null,\"bitgo_rest_api_root\":\"https://test.bitgo.com/api/v2\",\"bitgo_rest_api_access_token\":null,\"wallet_url_template\":\"https://blockchain.info/address/#{address}\",\"transaction_url_template\":\"https://blockchain.info/tx/#{txid}\"}', '2018-03-10 09:28:39', '2018-03-10 09:28:39')

This error due to character string not compatible with the existing collation.
So i add collation to database.yml
default: &default
  adapter: mysql2
  encoding: utf8
  collation: utf8_general_ci
  pool: 5
  host: <%= ENV['DATABASE_HOST'] %>
  username: <%= ENV['DATABASE_USER'] %>
  password: <%= ENV['DATABASE_PASS'] %>

After that re-run again setup. The insert from currencies.yml should be inserted properly.

 Currency Exists (0.3ms)  SELECT  1 AS one FROM `currencies` WHERE `currencies`.`code` = 'btc' LIMIT 1
  Currency Exists (0.4ms)  SELECT  1 AS one FROM `currencies` WHERE `currencies`.`code` = BINARY 'btc' LIMIT 1
  SQL (0.4ms)  INSERT INTO `currencies` (`id`, `code`, `key`, `symbol`, `base_factor`, `quick_withdraw_limit`, `options`, `created_at`, `updated_at`) VALUES (2, 'btc', 'satoshi', '฿', 100000000, 0.1, '{\"api_client\":\"BTC\",\"json_rpc_endpoint\":\"http://user:password@127.0.0.1:18332\",\"bitgo_test_net\":true,\"bitgo_wallet_id\":null,\"bitgo_wallet_address\":null,\"bitgo_wallet_passphrase\":null,\"bitgo_rest_api_root\":\"https://test.bitgo.com/api/v2\",\"bitgo_rest_api_access_token\":null,\"wallet_url_template\":\"https://blockchain.info/address/#{address}\",\"transaction_url_template\":\"https://blockchain.info/tx/#{txid}\"}', '2018-03-10 09:37:05', '2018-03-10 09:37:05')

### PR DESCRIPTION
**Pull request by ysv**

### CODE REVIEW

File: config/database.yml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 default: &default
   adapter: mysql2
   encoding: utf8
+  colation: <%= ENV.fetch('DATABASE_HOST', 'utf8_general_ci') %>
```
Comment (yivo) [inline config/database.yml]: @yssavchuk Interesting...

It also must no in application.yml. No *ing hidden defaults please!

**Review by yivo — CHANGES_REQUESTED** [review]

File: config/database.yml (hunk 1 above)
Comment (yivo) [reply config/database.yml]: Hey @yssavchuk 

I checked this again. We don't want ENV here. Hardcode please

**Review by yivo — APPROVED** [review]

File: config/database.yml [hunk 2]
```diff
@@ -1,6 +1,7 @@
 default: &default
   adapter: mysql2
   encoding: utf8
+  colation: utf8_general_ci
```
Comment (mod) [inline config/database.yml]: is it not a spelling mistake?

**Review by mod — CHANGES_REQUESTED** [review]

File: config/database.yml (hunk 2 above)
Comment (yivo) [reply config/database.yml]: @mod this is...

@yssavchuk could you pls update it?

**Review by mod — APPROVED** [review]

**Pull request merged**